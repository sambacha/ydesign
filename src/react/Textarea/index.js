import React from 'react';
import PropTypes from 'prop-types';
import { StyledInput } from '../Input';

const Textarea = React.forwardRef((props, ref) => (
  <StyledInput as={'textarea'} ref={ref} {...props} />
));

Textarea.defaultProps = {
  lineHeight: 'copy',
  height: 'auto',
};

Textarea.propTypes = {
  theme: PropTypes.object,
};

Textarea.displayName = 'Textarea';

export default Textarea;
