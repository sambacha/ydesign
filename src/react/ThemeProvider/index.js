import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
