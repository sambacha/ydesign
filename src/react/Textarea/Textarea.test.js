import Textarea from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Textarea component sanity', () => {
  it('has name', () => {
    expect(Textarea.displayName).toBe('Textarea');
  });

  it('matches default snapshot', () => {
    const component = render(<Textarea />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Textarea />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
