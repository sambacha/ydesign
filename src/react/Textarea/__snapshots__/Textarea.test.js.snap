// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.5;
  height: auto;
  color: #3F3D4B;
  background-color: #fff;
  font-size: 1rem;
  padding: 16px;
  border: 1px solid transparent;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c0:focus {
  outline: none;
  border-color: #4E3FCE;
}

.c0:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c0[type="color"] {
  min-width: 3rem;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.c0[type="color"]::-moz-color-swatch {
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.was-validated .c0:valid {
  border-color: #28C081;
}

.was-validated .c0:valid ~ svg.icon-valid {
  visibility: visible;
  color: #28C081;
}

.was-validated .c0:invalid {
  border-color: #DC2C10;
}

.was-validated .c0:invalid ~ svg.icon-invalid {
  visibility: visible;
  color: #DC2C10;
}

<input
  class="sc-pIWYI c0"
  color="text"
  font-size="1rem"
  height="auto"
/>
`;

exports[`Textarea component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.5;
  height: auto;
  color: #3F3D4B;
  background-color: #fff;
  font-size: 1rem;
  padding: 16px;
  border: 1px solid transparent;
  border-color: #ccc;
  border-radius: 4px;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c0:focus {
  outline: none;
  border-color: #4E3FCE;
}

.c0:disabled {
  opacity: 0.4;
  pointer-events: none;
}

.c0[type="color"] {
  min-width: 3rem;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
}

.c0[type="color"]::-webkit-color-swatch {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.c0[type="color"]::-moz-color-swatch {
  height: 2rem;
  width: 2rem;
  border: none;
  border-radius: 100%;
}

.was-validated .c0:valid {
  border-color: #28C081;
}

.was-validated .c0:valid ~ svg.icon-valid {
  visibility: visible;
  color: #28C081;
}

.was-validated .c0:invalid {
  border-color: #DC2C10;
}

.was-validated .c0:invalid ~ svg.icon-invalid {
  visibility: visible;
  color: #DC2C10;
}

<input
  class="sc-pIWYI c0"
  color="text"
  font-size="1rem"
  height="auto"
/>
`;
